### PR TITLE
feat: Executor tests - pre/post run scripts, postman/newman - container executor examples

### PR DESCRIPTION
**Pull request by tkonieczny**

### CODE REVIEW

**Review by nicufk — APPROVED** [review]

File: test/suites/special-cases/edge-cases-expected-fails.yaml [hunk 1]
```diff
@@ -58,3 +58,9 @@ spec:
   - stopOnFailure: false
     execute:
     - test: expected-fail-cypress-oomkilled
+  - stopOnFailure: false
+    execute:
+    - test: expected-fail-pre-run-script
+  - stopOnFailure: false
+    execute:
+    - test: expected-fail-post-run-script
```
Comment (rangoo94) [inline test/suites/special-cases/edge-cases-expected-fails.yaml:66]: Missing EOL on EOF

File: test/container-executor/executor-smoke/crd/postman.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+      uri: https://github.com/kubeshop/testkube
+      branch: main
+      path: test/postman/executor-tests/postman-executor-smoke.postman_collection.json
+      workingDir: test/postman/executor-tests
+  executionRequest:
+    args: ["run", "postman-executor-smoke.postman_collection.json", "--env-var", "TESTKUBE_POSTMAN_PARAM=TESTKUBE_POSTMAN_PARAM_value"]
+    preRunScript: "echo \"===== pre-run script\""
+    postRunScript: "echo \"===== post-run script\""
+    jobTemplate: "apiVersion: batch/v1\nkind: Job\nspec:\n  template:\n    spec:\n      containers:\n        - name: \"{{ .Name }}\"\n          image: {{ .Image }}\n          resources:\n            requests:\n              memory: 128Mi\n              cpu: 256m\n"
+    activeDeadlineSeconds: 180
```
Comment (rangoo94) [inline test/container-executor/executor-smoke/crd/postman.yaml:22]: Missing EOL on EOF

File: test/executors/container-executor-postman.yaml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+apiVersion: executor.testkube.io/v1
+kind: Executor
+metadata:
+  name: container-executor-postman-newman-6-alpine
+spec:
+  image: postman/newman:6-alpine
+  executor_type: container
+  types:
+  - container-executor-postman-newman-6-alpine/test
```
Comment (rangoo94) [inline test/executors/container-executor-postman.yaml:9]: Missing EOL on EOF

File: test/special-cases/edge-cases-expected-fails.yaml [hunk 4]
```diff
@@ -313,3 +313,49 @@ spec:
       - video=true
     jobTemplate: "apiVersion: batch/v1\nkind: Job\nspec:\n  template:\n    spec:\n      containers:\n        - name: \"{{ .Name }}\"\n          image: {{ .Image }}\n          resources:\n            limits:\n              memory: 30Mi\n              cpu: 1\n"
     activeDeadlineSeconds: 600
+---
+apiVersion: tests.testkube.io/v3
```
Comment (rangoo94) [inline test/special-cases/edge-cases-expected-fails.yaml:317]: Shouldn't it have `negativeTest: true`?

File: test/special-cases/edge-cases-expected-fails.yaml [hunk 5]
```diff
@@ -313,3 +313,49 @@ spec:
...
+      path: test/postman/executor-tests/postman-executor-smoke.postman_collection.json
+  executionRequest:
+    args:
+      - --env-var
+      - TESTKUBE_POSTMAN_PARAM=TESTKUBE_POSTMAN_PARAM_value
+    preRunScript: "echo \"===== pre-run script - EXPECTED FAIL\" && exit 128"
+    postRunScript: "echo \"===== post-run script\""
+    jobTemplate: "apiVersion: batch/v1\nkind: Job\nspec:\n  template:\n    spec:\n      containers:\n        - name: \"{{ .Name }}\"\n          image: {{ .Image }}\n          resources:\n            requests:\n              memory: 128Mi\n              cpu: 256m\n"
+---
+apiVersion: tests.testkube.io/v3
```
Comment (rangoo94) [inline test/special-cases/edge-cases-expected-fails.yaml:340]: Shouldn't it have `negativeTest: true`?

**Review by rangoo94 — APPROVED** [review]

File: test/special-cases/edge-cases-expected-fails.yaml (hunk 4 above)
Comment (rangoo94) [reply test/special-cases/edge-cases-expected-fails.yaml:317]: > No - we have separate ones. There are negative tests (with `-negative` names), and there are separate `expected-fail` tests to also have actual failures (that's needed to test different edge-cases).

File: test/special-cases/edge-cases-expected-fails.yaml (hunk 5 above)
Comment (rangoo94) [reply test/special-cases/edge-cases-expected-fails.yaml:340]: > No - we have separate ones. There are negative tests (with `-negative` names), and there are separate `expected-fail` tests to also have actual failures (that's needed to test different edge-cases).

File: test/special-cases/edge-cases-expected-fails.yaml (hunk 4 above)
Comment (tkonieczny) [reply test/special-cases/edge-cases-expected-fails.yaml:317]: no - there are negative tests (with `-negative` names), and there are separate `expected-fail` tests to also have actual failures (that's needed to test different edge-cases)
